src/GitHub.VisualStudio/GitHubPackage.cs

```diff
@@ -269,5 +296,20 @@ static ToolWindowPane ShowToolWindow(Guid windowGuid)
                 return sp.TryGetService(serviceType);
             }
         }
+
+        IVSGitExt CreateVSGitExt(string dteVersion)
+        {
+            // DTE.Version always ends with ".0" even for later minor versions.
+            switch (dteVersion)
+            {
+                case "14.0":
+                    return new Lazy<IVSGitExt>(() => new VSGitExt14(GetServiceAsync)).Value;
+                case "15.0":
+                    return new Lazy<IVSGitExt>(() => new VSGitExt15(GetServiceAsync)).Value;
+                default:
```

Comment:
The logic to construct the correct type should be in the VSGitExtDispatcher, so callers don't have to worry about it.
Also, since it needs to call GetServiceAsync on an IAsyncServiceProvider, it should take IAsyncServiceProvider in its constructor, so we know that that's a dependency that this type has. Later, when GitHubServiceProvider supports async calls, we can implement IAsyncServiceProvider on it and pass it in without major changes to the code.